protocol/shannon/config.go

```diff
@@ -60,11 +61,25 @@ type (
 		SessionTTL time.Duration `yaml:"session_ttl"`
 	}
 
+	ServiceFallback struct {
+		ServiceID      protocol.ServiceID `yaml:"service_id"`
+		SendAllTraffic bool               `yaml:"send_all_traffic"`
+		FallbackURLs   []string           `yaml:"fallback_urls"`
+	}
+
+	// ServiceFallbackConfig holds the fallback configuration for a service,
+	// including the endpoints and whether to send all traffic to fallback.
+	ServiceFallbackConfig struct {
```

Comment:
Do we have to make this public? It makes the code confusing since there is also a `ServiceFallback` struct.

Reply:
Yes, you're correct. This can be private and is better located in the `protocol.go` file. I've made this change.